### PR TITLE
Add support for scan link.

### PR DESCRIPTION
**Pull request by 3asm**
The PR adds support for the scan.
It also addresses 2 bugs. The 1st one is that guessing the file name will stop at the first which is often wrong.
The 2nd bug is an issue in `osv` not supporting `int` version.

An issue has been opened with `osv` project. Awaiting a fix, a hot patch of the content is done to make it compatible with `osv`.

### CODE REVIEW

File: tests/hotpatch_test.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import pathlib
+from agent import hotpatch
+
+
+def testHotPatch_whenComposerHasIntVersion_replacesWithString():
+    content = pathlib.Path("test/composer.lock").read_bytes()
+
+    _, patch_content = hotpatch.hotpatch("composer.lock", content)
+    assert '""version": "20190220",' in patch_content
```
Comment (rabsondev) [inline tests/hotpatch_test.py:9]:
```suggestion
import pathlib

from agent import hotpatch


def testHotPatch_whenComposerHasIntVersion_replacesWithString() -> None:
    content = pathlib.Path("test/composer.lock").read_bytes()

    _, patch_content = hotpatch.hotpatch("composer.lock", content)
    
    assert '""version": "20190220",' in patch_content
```

File: agent/osv_agent.py [hunk 2]
```diff
@@ -100,6 +121,21 @@ def _get_content(message: m.Message) -> bytes | None:
     content_url = message.data.get("content_url")
     if content_url is not None:
         return requests.get(content_url, timeout=60).content
+    content_url = _get_content_url(message)
+    if content_url is not None:
+        return content_url
+    return None
```
Comment (rabsondev) [inline agent/osv_agent.py:127]: Why not return `content_url` directly?

File: agent/hotpatch.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+"""Implementation of hotpatch due to known issues in OSV."""
+
+import re
+
+
+def hotpatch(path: str, content: bytes):
```
Comment (rabsondev) [inline agent/hotpatch.py:6]: Missing return type.

**Review by rabsondev — CHANGES_REQUESTED** [review]

File: agent/hotpatch.py (hunk 3 above)
Comment (3asm) [reply agent/hotpatch.py]: Fixed.

File: agent/osv_agent.py (hunk 2 above)
Comment (3asm) [reply agent/osv_agent.py:127]: Explicit is better than implicit.

**Review by rabsondev — APPROVED** [review]

**Pull request merged**